### DISCUSSION TITLE
Is it possible to make use of the STM32's low power modes using modm?

### DISCUSSION THREAD

**Discussion by bid-p** [modm-io/modm#1084] [Q&A] [answered]
Looking into lowering power consumption by entering the STM32 sleep, standby, stop modes, etc but the only documentation I've found online use STM's HAL. I'm curious if this or similar functionality is supported by modm?

**Comment by salkinium** [comment, marked as answer]
No, not at all. Of course you can manually enter these modes, but the entire architecture of modm is not designed for low-power operations. You could perhaps hack it, for example, the task scheduling is all based on polling, so you could enter WFI sleep mode in the main loop if you set the `modm:platform:systick` frequency to 1kHz, cos then you wake up at least every ms and run through the main loop once. But it's really a bad substitute for a proper RTOS which wakes up the right threads on an event.

Perhaps with the fiber scheduler, we could support low-power, but that's definitely not gonna happen anytime soon. I would recommend looking into a modern RTOS like Zephyr for low-power operations instead or perhaps something completely different like [RTIC.rs](https://rtic.rs/2/book/en/). Generally though, really low-power (10y on a coin cell) is going to require a whole system approach and you need to measure and fine tune your whole application including non-microcontroller hardware.

**Reply by bid-p** [reply]
Understood, thanks for your response!
